### ISSUE TITLE
mpv gets OpenGL version and capabilities wrong under Mesa/radeon (r600g)

### ISSUE THREAD

**Issue by v-fox** [mpv-player/mpv#2938]
Some info  about my system:
`glxinfo`:
```
...
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: X.Org (0x1002)
    Device: AMD BARTS (DRM 2.43.0, LLVM 3.7.0) (0x6738)
    Version: 11.1.2
    Accelerated: yes
    Video memory: 1024MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 3.3
    Max compat profile version: 3.0
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.0
OpenGL vendor string: X.Org
OpenGL renderer string: Gallium 0.4 on AMD BARTS (DRM 2.43.0, LLVM 3.7.0)
OpenGL core profile version string: 3.3 (Core Profile) Mesa 11.1.2
OpenGL core profile shading language version string: 3.30
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
...
OpenGL version string: 3.0 Mesa 11.1.2
OpenGL shading language version string: 1.30
OpenGL context flags: (none)
...
OpenGL ES profile version string: OpenGL ES 3.0 Mesa 11.1.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00
```

So in mpv I get something like:
```
...
[vo/opengl-hq] Initializing OpenGL backend 'x11'
[vo/opengl-hq/x11] X11 opening display: :0
[vo/opengl-hq/x11] X11 running at 1920x1080 (":0" => local display)
[vo/opengl-hq/x11] Display 0 (DVI-0): [0, 0, 1920, 1080] @ 60.000000 FPS
[vo/opengl-hq/x11] Current display FPS: 60.000000
[vo/opengl-hq] GLX chose FB config with ID 0x6060af50
[vo/opengl-hq] GLX chose visual with ID 0x334
[vo/opengl-hq] GL_VERSION='3.0 Mesa 11.1.2'
[vo/opengl-hq] Detected desktop OpenGL 3.0.
[vo/opengl-hq] GL_VENDOR='X.Org'
[vo/opengl-hq] GL_RENDERER='Gallium 0.4 on AMD BARTS (DRM 2.43.0, LLVM 3.7.0)'
[vo/opengl-hq] GL_SHADING_LANGUAGE_VERSION='1.30'
[vo/opengl-hq] Loaded functions for 210/builtin.
[vo/opengl-hq] Loaded functions for 210/builtin.
[vo/opengl-hq] Loaded functions for 300/builtin.
[vo/opengl-hq] Loaded functions for 210/builtin.
[vo/opengl-hq] Loaded functions for 110/GL_EXT_unpack_subimage.
[vo/opengl-hq] Loaded functions for 300/GL_ARB_framebuffer_object.
[vo/opengl-hq] Loaded functions for 300/GL_ARB_vertex_array_object.
[vo/opengl-hq] Loaded functions for 300/GL_ARB_texture_rg.
[vo/opengl-hq] Loaded functions for 320/GL_ARB_sync.
[vo/opengl-hq] Loaded functions for 0/GLX_SGI_swap_control.
[vo/opengl-hq] Loaded functions for 0/GLX_SGI_video_sync.
[vo/opengl-hq] Loaded functions for 0/GL_NV_vdpau_interop.
[vo/opengl-hq] Loaded functions for 430/GL_ARB_debug_output.
[vo/opengl-hq] Loaded functions for 310/GL_ARB_uniform_buffer_object.
...
[vo/opengl-hq] Disabling NNEDI3 (OpenGL 3.3 required).
...
[vo/opengl-hq] Assuming 60.000000 FPS for display sync.
```
So to use nnedi3 I have to force OpenGL ES and only in "shader" mode. "ubo" mode is broken with artifacts.

Also, if I use x11egl backend I get this:
```
...
[vo/opengl-hq] Loaded functions for 210/builtin.
[vo/opengl-hq] Loaded functions for 210/builtin.
[vo/opengl-hq] Loaded functions for 300/builtin.
[vo/opengl-hq] Loaded functions for 210/builtin.
[vo/opengl-hq] Loaded functions for 110/GL_EXT_unpack_subimage.
[vo/opengl-hq] Loaded functions for 300/GL_ARB_framebuffer_object.
[vo/opengl-hq] Loaded functions for 300/GL_ARB_vertex_array_object.
[vo/opengl-hq] Loaded functions for 300/GL_ARB_texture_rg.
[vo/opengl-hq] Loaded functions for 320/GL_ARB_sync.
[vo/opengl-hq] Loaded functions for 0/GL_NV_vdpau_interop.
[vo/opengl-hq] Loaded functions for 430/GL_ARB_debug_output.
[vo/opengl-hq] Loaded functions for 310/GL_ARB_uniform_buffer_object.
...
[vo/opengl-hq] swap_control extension missing.
```
And no interpolation even thought `glxinfo -s | grep -E 'swap|sync'`:
```
    GLX_INTEL_swap_event
    GLX_OML_swap_method
    GLX_SGI_swap_control
    GLX_INTEL_swap_event
    GLX_MESA_swap_control
    GLX_OML_swap_method
    GLX_OML_sync_control
    GLX_SGI_swap_control
    GLX_SGI_video_sync
    GLX_INTEL_swap_event
    GLX_MESA_swap_control
    GLX_OML_swap_method
    GLX_OML_sync_control
    GLX_SGI_swap_control
    GLX_SGI_video_sync
    GL_ARB_sync
    GL_ARB_sync
    GL_OES_EGL_sync
```
And that with "TearFree" Xorg option which is: "Enable tearing prevention using the hardware page flipping mechanism."

"ubo" mode of needi3 may be a Mesa bug but everything else is clearly on mpv. I'm also affected by issue #1814.

**Comment by wm4** [comment]
The problem is probably that mpv request a 3.0 context, which always implies a compatibility context, and the driver probably doesn't allow newer GL versions on a compatibility context.

Although mpv doesn't need the compatibility context, it still requests 3.0 for compatibility with Intel Mesa.

**Comment by v-fox** [comment]
Yeah, which is weird since even these [OpenGL specs](https://www.opengl.org/wiki/Core_And_Compatibility_in_Contexts) say:
>The 3.0 form of context creation did not have the concept of profiles. There was only one form of OpenGL: the core specification. In 3.2, OpenGL was effectively split into two profiles, core and compatibility. **An implementation was only required to define core, so compatibility is not guaranteed to be available**. However, on most implementations, the compatibility profile will be available. 

So, compatibility profile isn't guaranteed to exist or be of any specific version. And yet its usage is a [common issue](https://github.com/mosra/magnum/issues/80#issuecomment-145958456).

**Comment by wm4** [comment]
We don't need the compatibility profile. But Intel Mesa (at least older hw/drivers) provides only up to OpenGL 3.0.

**Comment by v-fox** [comment]
Then why is it forced on everything else and all while >3.0 features exist ? Besides, [Mesa's Intel support is 3.3](https://mesamatrix.net) and for [more than a year now](https://www.phoronix.com/scan.php?page=news_item&px=MTc5MjY). The 12th release will probably be with 4.2 or 4.3.

x11egl's ignorance to vsync is no less baffling.

**Comment by wm4** [comment]
This is getting annoying.

>x11egl's ignorance to vsync is no less baffling.

Do you even have a clue?

**Comment by wm4** [comment]
This was probably fixed with 3353923f2906368fb4d0b1fc0927f9f8fce30cf8. (Not the swap_control thing, because it's default on in EGL, and also useless.)
